### PR TITLE
apply overwrite fragments

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public override string Name => nameof(ApplyOverwriteFragments);
+
+        public override int BuildOrder => 0x08;
+
+        public override void Postbuild(ImmutableList<FileModel> models, IHostService host)
+        {
+            var overwriteApplier = new OverwriteApplier(host, OverwriteModelType.MarkdownFragments);
+            foreach (var fileModel in models)
+            {
+                var overwriteDocumentModels = fileModel.MarkdownFragmentsModel?.ModelWithCache.Content as List<OverwriteDocumentModel>;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs]: can use my newly added `MarkdownFragmentsModel.Content` ...

File: src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    using Microsoft.DocAsCode.Plugins;
+
+    [Export(nameof(SchemaDrivenDocumentProcessor), typeof(IDocumentBuildStep))]
+    public class ApplyOverwriteFragments : BaseDocumentBuildStep, ISupportIncrementalBuildStep
+    {
+        public override string Name => nameof(ApplyOverwriteFragments);
+
+        public override int BuildOrder => 0x08;
+
+        public override void Postbuild(ImmutableList<FileModel> models, IHostService host)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs]: >Postbuild [](start = 29, length = 9)

can move into `build` step?

File: src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    {
+                        var uidDefiniton = fileModel.Uids.Where(s => s.Name == overwriteDocumentModel.Uid).ToList();
+                        if (uidDefiniton.Count == 0)
+                        {
+                            Logger.LogWarning($"Unable to find UidDefinition for Uid {overwriteDocumentModel.Uid}");
+                        }
+
+                        if (uidDefiniton.Count > 1)
+                        {
+                            Logger.LogWarning($"There are more than one UidDefinitions found for Uid {overwriteDocumentModel.Uid}");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs]: >LogWarning [](start = 35, length = 10)

print out the detailed source line number for what uid definition is duplicate

File: src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Build.Common;
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    [Export(nameof(SchemaDrivenDocumentProcessor), typeof(IDocumentBuildStep))]
+    public class ApplyOverwriteFragments : BaseDocumentBuildStep, ISupportIncrementalBuildStep
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs]: >ApplyOverwriteFragments [](start = 17, length = 23)

add ut for overwrite & incremental overwrite

Comment (928PJY) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/ApplyOverwriteFragments.cs]: UT will be added after Renze's next PR

---
In reply to: [166499334](https://github.com/dotnet/docfx/pull/2419#discussion_r166499334) [](ancestors = 166499334)

**Pull request merged**